### PR TITLE
Feature/directives

### PR DESCRIPTION
**Pull request by kyunglee1**
# Checklist

- [ ] Bugfix
- [x] New feature
- [ ] Refactor

# Related Issue

 - Lack of support for GraphQL variables (multiple) and directives (single @skip/@include)

# Solution

 - Add destructuring support for inbound query strings containing variables (multiple) and directives (single @skip/@include). 
 - Parse and replace query string variables with their corresponding values. 
 - Parse and replace query string directives, keeping or omitting the field based on the value of the variable in the directive.
 - Pass in GraphQL variables (body.variables) to the server-cache read/write functions.

# Additional Info

  - Server-cache read/write operations account for variables/directives. Browser-cache still does not currently support them.
  - Multiple directives (both @skip & @include) are currently not supported.

### CODE REVIEW

**Review by raymondcodes — APPROVED** [review]
Approving new features

File: src/lfuBrowserCache.js [hunk 1]
```diff
@@ -136,6 +136,7 @@ LFUCache.prototype.read = async function (queryStr) {
         arrayHashes,
         queries[query].fields
       );
+      console.log('reaches lfucache func here');
```
Comment (pjmsullivan) [inline src/lfuBrowserCache.js:139]: Do we need console logs here?

**Review by cssim22 — CHANGES_REQUESTED** [review]
Please remove eslint, and files that are not being changed from the pull request

**Review by pjmsullivan — CHANGES_REQUESTED** [review]
Delete server file and extraneous console log

**Review by raymondcodes — CHANGES_REQUESTED** [review]
Requesting changes with unnecessary files and eslint

**Review by pjmsullivan — APPROVED** [review]
Your changes look great!  Excellent work.

**Review by cssim22 — APPROVED** [review]
Everything looks great. Nice work!

**Review by raymondcodes — APPROVED** [review]
Approved requested changes

**Pull request merged**